client/state/posts/selectors.js

```diff
@@ -41,6 +42,23 @@ export function getSitePost( state, siteId, postId ) {
 }
 
 /**
+ * Returns an array of post objects by site ID.
+ *
+ * @param  {Object} state  Global state tree
+ * @param  {Number} siteId Site ID
+ * @return {?Array}         Site posts
+ */
+export function getSitePosts( state, siteId ) {
```

Comment:
Question here - Posts/Pages are commingled in this part of the state tree correct?  Would there be any value to providing a filter here for `type`?  I suppose it could be added down the line as well.